src/snowflake/snowpark/mock/_functions.py

```diff
@@ -220,6 +220,39 @@ def mock_avg(column: ColumnEmulator) -> ColumnEmulator:
     return ColumnEmulator(data=[res], sf_type=ColumnType(res_type, False))
 
 
```

Comment:
used here basically the `avg` implementation and adjusted accordingly. Tests should be done as part of merge back